fhirstore/search/corequerybuilder.py

```diff
@@ -0,0 +1,139 @@
+            element_query["query_string"]["query"] = value
+        elif string_modif:
+            string_modifier = string_modif.group(2)
+            string_field = string_modif.group(1)
+            if string_modifier == "contains":
+                element_query["query_string"]["query"] = f"*{value}*"
+                element_query["query_string"]["default_field"] = string_field
+
+            elif string_modifier == "exact":
+                element_query["query_string"]["query"] = value
+                element_query["query_string"]["fields"] = [string_field]
+
+            elif string_modifier == "not":
+                element_query["bool"]["must_not"]["match"][string_field] = f"{value}"
+
+            elif string_modifier == "not-in":
+                element_query["simple_query_string"]["query"] = f"-{value}"
+                element_query["simple_query_string"]["fields"] = [string_field]
+
+            elif string_modifier == "in":
+                element_query["query_string"]["query"] = f"{value}"
+
+            elif string_modifier == "below":
+                element_query["simple_query_string"]["query"] = f"({value})*"
+                element_query["simple_query_string"]["fields"] = [string_field]
+            elif string_modifier == "identifier":
+                element_query["simple_query_string"]["query"] = f"{value}"
+                element_query["simple_query_string"]["fields"] = [
+                    f"{string_field}.identifier.value"
+                ]
+
+        elif numeric_prefix:
+            element_query["range"][key] = {
+                number_prefix_matching[numeric_prefix.group(1)]: numeric_prefix.group(2)
+            }
+        elif eq_prefix:
+            element_query["match"][key] = eq_prefix.group(2)
+
+        elif special_prefix:
+            if special_prefix.group(1) == "ne":
+                element_query["simple_query_string"]["query"] = f"-{special_prefix.group(2)}"
+                element_query["simple_query_string"]["fields"] = [key]
+
+        elif pipe_suffix:
+            system, code = re.split(r"\|", value)
+            element_query["bool"]["must"] = [{"match": {f"{key}.system": system}}]
+            element_query["bool"]["must"].append(
+                {"simple_query_string": {"query": code, "fields": [f"{key}.code", f"{key}.value"],}}
+            )
+
+        elif isinstance(value, str):
+            element_query["simple_query_string"]["query"] = f"({value})*"
+            element_query["simple_query_string"]["fields"] = [key]
+
+    return element_query
+
+
+# This class transforms a dictionary of fields, and values into a dictionary
+# which can be the body of an elasticsearch query
+
+
+class CoreQueryBuilder:
+    def __init__(self):
+        self.query = {}
+
+    def validate_parameters(self, core_args: SearchArguments):
+        """Validates that parameters is in dictionary form
+        """
+        assert isinstance(core_args, Mapping), "parameters must be a dictionary"
```

Comment:
assert isinstance(core_args, dict), then? Or maybe it's because of defaultdict that you use Mapping?